### DISCUSSION TITLE
Ability to re-use permission sets between roles

### DISCUSSION THREAD

**Discussion by kropsi** [directus/directus#7032] [Feature Requests]
We have a case where we want to allow different organizations to access the CMS under organization-specific roles, but they have to have access only to their own created items and files. The idea is to create a role for each organization and configure the necessary permissions. All organizations must have the same permissions, which should be synchronized when something changes. It seems that now you will have to go and do it manually for each organization.

It could be helpful to re-use permission sets between different roles or have an option to export and import permissions.

**Comment by benhaynes** [comment]
Could happen in one of two ways:

* An option for a role to "mirror" or reference the permissions of another role.
* An option to duplicate / copy the permissions from one to another (not synced long-term)

I think both of these are useful to have.

**Reply by uebmaster** [reply]
this is the same that #2781 ?

**Reply by kropsi** [reply]
@uebmaster it is not the same. I agree with @benhaynes that mirroring permissions is a better approach. Changing permissions for multiple roles of the same type of entity, f.ex. the organization is probably even a bigger pain. I would say that roles and permissions must be split. So you can have multiple roles, and choose which set of permissions they should have. This way when you change permissions one place they will reflect on multiple roles.

**Comment by Pachat** [comment]
Meanwhile we get an integrated way to copy permissions from a role to another, I use the following brute force method. If
`bc8ccc27-25ae-4200-9c79-6658cf1a19b5` is the Id of the target role and
`06a93f09-9d51-428b-93a4-f2b71bd25ced` the Id of the source role as found in the `directus_roles` collection.
I delete the permissions of the target role:
`DELETE FROM directus_permissions WHERE role="bc8ccc27-25ae-4200-9c79-6658cf1a19b5";`
and recreate them from the source role:
`INSERT INTO directus_permissions ("action",collection,fields,permissions,presets,role,validation) SELECT"action",collection,fields,permissions,presets,"bc8ccc27-25ae-4200-9c79-6658cf1a19b5",validation FROM directus_permissions WHERE role="06a93f09-9d51-428b-93a4-f2b71bd25ced";
`

Recommendations:

1. Obvious: Have backup!
2. Stop Directus
3. Prefer a direct method (for sqlite I used the CLI: `sqlite3 data.db "SQL statement"`)
4. Do some cleaning (for sqlite I used: `sqlite3 data.db "VACUUM"`)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
